src/libraries/Common/src/System/IO/FileSystem.DirectoryCreation.Windows.cs

```diff
@@ -83,10 +93,9 @@ public static unsafe void CreateDirectory(string fullPath, byte[]? securityDescr
                     lpSecurityDescriptor = (IntPtr)pSecurityDescriptor
                 };
 
-                while (stackDir.Count > 0)
+                for (int i = count - 1; 0 <= i; i--)
```

Comment:
```suggestion
                for (int i = count - 1; i >= 0; i--)
```
Don't use yoda conditions here.